### DISCUSSION TITLE
[Feature Request]: ImagePrompt, PyroCanny, CPDS in Log.

### DISCUSSION THREAD

**Discussion by DavidKouw** [lllyasviel/Fooocus#2540] [Ideas]
[Feature Request]: History log update. Section in HTML log where ImagePrompt, PyroCanny, CPDS, FaceSwap parameters that were used during image generation will be specified. It is often necessary to select very precise settings, and it can be extremely problematic to remember them, I would be very grateful for adding this data to the log. Is this possible?

**Comment by mashb1t** [comment]
This would be possible but images used in image input will not be restored as they're not saved. This is also why metadata and history log do not include any image prompt parameters.

**Reply by DavidKouw** [reply]
Of course, I understand that. In my case, I often use pose blanks through Canny and CDPS, and their settings interact differently with the models. It's not hard for me to remember which image I used to generate them, but remembering dozens of attempts to find the right combination of settings is a bit frustrating. So I would be very grateful for such information in the log file. Thanks

**Reply by ILikeToasters** [reply]
I can see that being useful. I have often had to take screenshots of my settings when I find what works and plan on returning to using them again.
